create-custom-map-style/README.md

```diff
@@ -0,0 +1,471 @@
+1. Start up Maputnik
+
+    ```bash
+
+    ./maputnik
+    ```
+
+ And open your browser to <http://localhost:8000/#12.44/40.7356/-74.0541>
+
+1. It would be best to start with a clean canvas. Click "Open" in the top navigation bar, then scroll down to the bottom of "Gallery Styles" and selected "Empty Style".
+
+    ![Open](media/MaputnikOpen.png)
+
+    *Open A Style*
+
+    ![Empty](media/MaputnikEmpty.png)
+
+    *Empty Canvas*
+
+1. Tessera is a pluggable [map tile](https://en.wikipedia.org/wiki/Tiled_web_map) server. Using the power of the [tilelive](https://github.com/mapbox/tilelive) ecosystem, it is capable of serving and rendering map tiles from many sources. To stream data from most sources you can install the tilelive providers as npm packages. For an Amazon Location Service source we use tilelive-aws.
+1. From your terminal run
+
+    ```bash
+
+    mkdir tileserver
+    cd tileserver
+    ```
+
+1. We'll need to init a new node project in the current directory
+
+    ```bash
+
+    $ npm init --yes
+
+    Wrote to ... /create-custom-map-style/tileserver/package.json:
+
+    {
+      "name": "tileserver",
+      "version": "1.0.0",
+      "description": "",
+      "main": "index.js",
+      "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1"
+      },
+      "keywords": [],
+      "author": "",
+      "license": "ISC"
+    }
+    ```
+
+1. Next we need to add the Amazon Location Service API as a data source. However, in order to do this we need to use a local proxy that can handle SigV4 (TL;DR a way to provide your AWS credentials with http requests). For this we will use [tessera](https://github.com/mojodna/tessera) and [tilelive-aws](https://github.com/beatleboy501/tilelive-aws). From a new terminal (the `maputnik` process will be occupying the other terminal) run the following:
+
+    ```bash
+
+    nvm use 15
+    npm i tessera tilelive-aws
+    ```
+
+1. Presuming you have the standard AWS environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_REGION`) set correctly, this should be enough to get started using `tessera`. You'll need to provide the full file path to your module, so from your terminal run:
+
+    ```bash
+
+    node_modules/tessera/bin/tessera -r tilelive-aws aws:///<YOUR AMAZON LOCATION SERVICE MAP NAME>
+    ```
+
+    You can also add this as a script to your `package.json` file to save time on subsequent invocations.
+
+1. You should see some output in sdtout like `Listening at http://0.0.0.0:8080`.
+
+1. Now go back to Maputnik in your browser and click "Data Sources" in the top navigation bar. There should not be any active data sources. If you see any be sure to delete them.
+
+    ![Data Sources](media/MaputnikDataSources.png)
+
+    *Data Sources*
+
+1. We're going to add a new data source as shown below.
+
+- `Source ID`: `esri`
+- `Source Type`: `Vector (XYZ URLs)`
+- `1st Tile URL`: `http://localhost:8080/{z}/{x}/{y}.pbf`
+- `Min Zoom`: `0`
+- `Max Zoom`: `22`
```

Comment:
Potential simplification now that we've got `tilelive-aws` sorted out: this can now be a TileJSON source, with the URL `http://localhost:8080/index.json`.